src/scripts/main.js

```diff
@@ -1,3 +1,24 @@
 'use strict';
 
 // write code here
+const tree = document.querySelector('.tree');
+
+for (const li of tree.querySelectorAll('li')) {
+  const span = document.createElement('span');
+
+  li.prepend(span);
+  span.append(span.nextSibling);
+
+  tree.onclick = function (e) {
+    if (e.target.tagName !== 'SPAN') {
+      return;
+    }
+
+    const childrenContainer = e.target.parentNode.querySelector('ul');
+
+    if (!childrenContainer) {
+      return;
+    }
+    childrenContainer.hidden = !childrenContainer.hidden;
+  };
+}
```

Comment:
The `tree.onclick` event handler is defined inside the `for` loop, which means it will be redefined for every `li` element in the tree. This is inefficient and could lead to performance issues. Consider moving the `tree.onclick` event handler outside of the loop to ensure it is only defined once.